Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2882,3 +2892,37 @@ public extension Tensor where Scalar: Numeric {
         matmul(lhs, rhs)
     }
 }
+
+//===------------------------------------------------------------------------------------------===//
+// Precondition helpers.
+//===------------------------------------------------------------------------------------------===//
+
+internal extension Tensor {
+    @usableFromInline
+    func preconditionCheck(
+        axes: Tensor<Int32>,
```

Comment:
More of a general question than a comment on this change: do we end up with these two overloads (on `Tensor<Int32>/[Int]` of many functions?  Should we be generic over `Collection where Element : BinaryInteger` or something?

Reply:
Perhaps, but I'm currently unconvinced. For how, I'll leave this as-is, as this follows the current existing patterns.

Reply:
Well of course follow the existing patterns until we make a more general decision.

Reply:
SGTM; thanks!